hail/src/main/scala/is/hail/expr/ir/ComputeUsesAndDefs.scala

```diff
@@ -0,0 +1,98 @@
+package is.hail.expr.ir
+
+import scala.collection.mutable
+
+case class UsesAndDefs(uses: Memo[mutable.Set[RefEquality[Ref]]], defs: Memo[BaseIR])
+
+object ComputeUsesAndDefs {
+  def apply(ir0: BaseIR, freeVariablesError: Boolean = true): UsesAndDefs = {
```

Comment:
In NormalizedNames, you called this `allowFreeVariables`.  Consistent sense seems nicer (and safer).